diff --git a/arch/riscv/kernel/setup.c b/arch/riscv/kernel/setup.c
index a990a6cb184f..1980b9c939f1 100644
--- a/arch/riscv/kernel/setup.c
+++ b/arch/riscv/kernel/setup.c
@@ -41,10 +41,9 @@ unsigned long boot_cpu_hartid;
 
 void __init parse_dtb(void)
 {
-	if (early_init_dt_scan(dtb_early_va))
-		return;
+	if (!early_init_dt_scan(dtb_early_va))
+    pr_err("No DTB passed to the kernel\n");
 
-	pr_err("No DTB passed to the kernel\n");
 #ifdef CONFIG_CMDLINE_FORCE
 	strlcpy(boot_command_line, CONFIG_CMDLINE, COMMAND_LINE_SIZE);
 	pr_info("Forcing kernel command line to: %s\n", boot_command_line);
